underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -574,6 +580,24 @@ private boolean deleteInternal(String key) {
   }
 
   /**
+   * Waits until the given key exists or the timeout is exceeded. This should only be used when
+   * the key is expected to exist.
+   *
+   * @throws IOException if the timeout is exceeded
+   */
+  private void ensureExists(String key, long timeoutMs) throws IOException {
+    long startMs = System.currentTimeMillis();
+     do {
+      if (exists(key)) {
+        return;
+      } else {
+        CommonUtils.sleepMs(Constants.SECOND_MS);
+      }
+    } while (System.currentTimeMillis() - startMs < timeoutMs);
```

Comment:
I've changed the code to be more similar. `waitFor` doesn't allow setting of the retry period which should be a much longer period of time in this case.